core/unix/module_elf.c

```diff
@@ -209,6 +211,52 @@ elf_dt_abs_addr(ELF_DYNAMIC_ENTRY_TYPE *dyn, app_pc base, size_t size, size_t vi
     return tgt;
 }
 
+/* TODO: Is there a way of accessing elf_loader_t's load_base and image_size here?
+ * If so, then this function may not be needed.
+ */
+static size_t
+module_map_file(os_module_data_t *os_mod_data)
+{
+    int fd;
+    uint64 size;
+
+    fd = open_syscall(os_mod_data->source_file_name, O_RDONLY, 0);
+    ASSERT(fd != -1);
+
+    if (!os_get_file_size_by_handle(fd, &size))
+        ASSERT(false);
+
+    os_mod_data->source_file_map =
+        (byte *)mmap_syscall(NULL, size, PROT_READ, MAP_PRIVATE, fd, 0);
+    ASSERT(os_mod_data->source_file_map != (byte *)-1);
+
+    return size;
+}
+
+/* Extract .dynstr file offset from section header based on flat mmap of ELF file */
+static Elf64_Off
+module_get_dynstr_offset(app_pc base, size_t size)
```

Comment:
I think `module_get_section_with_name(base, size, ".dynstr")` is exactly the same, except returning `sh_addr`?  Plus it has a few more sanity checks.  There's no caller of it today, so I would say let's change it to `sh_offset`, which makes more sense anyway for what it's doing.  Then we don't need this new function.

Reply:
Good spot! Done.